src/main/scala/castalia/matcher/RequestMatcherActor.scala

```diff
@@ -0,0 +1,58 @@
+package castalia.matcher
+
+import akka.actor._
+import akka.http.scaladsl.model.HttpRequest
+import castalia.matcher.MatcherActor.RespondIfMatched
+import castalia.matcher.RequestMatcherActor.{AddMatcher, FindMatchAndForward}
+import castalia.matcher.types._
+
+/**
+  * Created by m06f791 on 4-2-2016.
+  */
+
+object RequestMatcherActor {
+
+  case class FindMatchAndForward(httpRequest: HttpRequest, origin: ActorRef)
+
+  case class AddMatcher(segments: Segments, handler: ActorRef)
+
+  def props(): Props = Props(new RequestMatcherActor with RequestMatcherActorCreatorImplementation)
+}
+
+class RequestMatcherActor extends Actor with ActorLogging {
+  this: RequestMatcherActorCreator =>
+
+  def receive: Receive = normal(Set.empty[(ActorRef, Segments)])
+
+  def normal(matchers: Set[(ActorRef, Segments)]): Receive = {
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Default case not specified.](https://www.codacy.com/app/scala-academy/Castalia/file/1993124807/issues/source?bid=2998213&fileBranchId=3120054#l27)